### DISCUSSION TITLE
version 0.23.3: Did something change for the avatar field in users?

### DISCUSSION THREAD

**Discussion by Gys** [pocketbase/pocketbase#5993] [Q&A] [answered]
The code below does not work for version 0.23.3 but did work before in 0.22.x (with slight changes because of SDK changes).

The line `app.Save(user)` now results in the error `avatar: Invalid new files: [avatar.jpg]..`

Did something change for the avatar field? I think in version 0.22.x this is a text field without any checks for valid file(s)?

````
func TestAddAvatar(t *testing.T) {
	app, err := tests.NewTestApp(TEST_DATA_DIR)
	if err != nil {
		t.Fatal(err)
	}
	defer app.Cleanup()

	users, err := app.FindCollectionByNameOrId("users")
	if err != nil {
		t.Fatal(err)
	}
	user := core.NewRecord(users)

	user.Set("username", "testuser")
	user.Set("email", "test@test.com")
	user.Set("password", "testtestest")
	user.Set("passwordConfirm", "testtestest")

	fileName := "avatar.jpg"
	imageData := []byte{0xFF, 0xD8, 0xFF, 0xE0} // Basic JPEG header
	filePath := filepath.Join(user.BaseFilesPath(), fileName)
	fs, err := app.NewFilesystem()
	if err != nil {
		t.Fatalf("failed to create filesystem")
	}
	defer fs.Close()

	newFile, err := filesystem.NewFileFromBytes(imageData, fileName)
	if err != nil {
		t.Fatal("failed to create fileFromBytes")
	}
	if err := fs.UploadFile(newFile, filePath); err != nil {
		t.Fatalf("failed to upload file: %v", err)
	}

	user.Set("avatar", fileName)
	if err := app.Save(user); err != nil {
		t.Fatal(err)
	}
}
````

**Comment by ganigeorgiev** [comment, marked as answer]
Record file operations were normalized and new uploaded files can be added as `*filesystem.File` instance (see https://pocketbase.io/docs/go-records/#create-new-record).

If you want to upload the file manually as in your example you'll have to save the record without validations but I'd advice against it. Instead you can change the above code to something like:

```go
func TestAddAvatar(t *testing.T) {
    app, err := tests.NewTestApp(TEST_DATA_DIR)
    if err != nil {
        t.Fatal(err)
    }
    defer app.Cleanup()

    users, err := app.FindCollectionByNameOrId("users")
    if err != nil {
        t.Fatal(err)
    }

    user := core.NewRecord(users)
    user.Set("username", "testuser")
    user.Set("email", "test@test.com")
    user.Set("password", "testtestest")

    imageData := []byte{0xFF, 0xD8, 0xFF, 0xE0} // Basic JPEG header

    newFile, err := filesystem.NewFileFromBytes(imageData, "avatar.jpg")
    if err != nil {
        t.Fatal("failed to create fileFromBytes")
    }

    user.Set("avatar", newFile)

    err = app.Save(user)
    if err != nil {
        t.Fatal(err)
    }
}
```

**Reply by Gys** [reply]
Thank you for the very quick reply!

So as a side effect of the normalization, the avatar filename is now determined by the system, instead of by me.

**Reply by ganigeorgiev** [reply]
You can still set a custom name exactly as it is if you really want to by replacing `newFile.Name = newFile.OriginalName` but I don't recommend it unless your use case requires it because it is possible to end up colliding/replacing an existing file and you'll also will have to take care for manually sanitizing the name if it comes from untrusted user input.
